recipes/nvcloth/1.1.6/conanfile.py

```diff
@@ -30,24 +35,62 @@ class NvclothConan(ConanFile):
         "use_dx11": False
     }
 
-    generators = "cmake"
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-    
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
-    
+        copy(self, "CMakeLists.txt", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.settings.os in ["Windows", "Macos"]:
+            # static builds are not supported
+            self.options.rm_safe("shared")
+            self.package_type = "shared-library"
```

Comment:
```suggestion
```

Please, do not change the package-type in case both type are supported in any OS.